### PR TITLE
[CORL-2787] quick site search

### PR DESCRIPTION
**Pull request by marcushaddon**
## What does this PR do?
This PR makes it possible for moderators to search by site name while filtering comments by site in the moderation view.

## These changes will impact:

- [ ] commenters
- [x] moderators
- [x] admins
- [ ] developers

## What changes to the GraphQL/Database Schema does this PR introduce?
None.

## Does this PR introduce any new environment variables or feature flags?
No.

## If any indexes were added, were they added to `INDEXES.md`?
No indexes were added or modified, but if you have not added a text index on sites as indicated in INDEXES.md, you will need to do that

(from INDEXES.md)
## 2021-12-17

```
db.sites.createIndex({ tenantID: 1, "$**": "text", createdAt: -1 });
```
  - This index creates a text search on sites so that when you use a text-filter base on site name, Mongo knows how to resolve your search query.
---

## How do I test this PR?
1. (Not required but helpful) create lots of sites
2. As a user with moderation privileges navigate to the moderation queue
3. Click on the Site Selector component in the upper left hand corner
4. Observe that the dropdown opens as usual, and you can scroll through sites and it paginates as needed
5. Observe that the button has turned into a text input
6. Type the first few letters of a site name, wait one second
7. Observe that the dropdown updates and is populated only by sites who's names begin with that prefix
8. Select a site from the dropdown and observe that the menu closes, that site's name is displayed on the button, and comments are filtered to just comments from that site

## Where any tests migrated to React Testing Library?
No.

## How do we deploy this PR?
No special considerations needed.

### CODE REVIEW

File: src/core/client/admin/components/AutoLoadMore.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable */
```
Comment (kabeaty) [inline src/core/client/admin/components/AutoLoadMore.tsx]: Can you please remove?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 2]
```diff
@@ -10,18 +17,20 @@ import {
...
-  Button,
   ClickOutside,
   Dropdown,
   Flex,
   Popover,
   Spinner,
+  TextField,
 } from "coral-ui/components/v2";
 
+import { Localized } from "@fluent/react/compat";
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: Can you please move this import up to the correct grouping?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 3]
```diff
@@ -41,13 +51,31 @@ const PaginatedSelect: FunctionComponent<Props> = ({
   selected,
   className,
 }) => {
+  const filterRef = useRef<HTMLInputElement>(null);
+  const [open, setOpen] = useState(false);
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: What about using `useToggleState` here instead? This hook is made for situations like this and includes the ability to use something like `isPopoverVisible`, `setIsPopoverVisible`, and `togglePopoverVisibility`.

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 4]
```diff
@@ -41,13 +51,31 @@ const PaginatedSelect: FunctionComponent<Props> = ({
...
+
+  useEffect(() => {
+    if (open && filterRef.current) {
+      filterRef.current.focus();
+    }
+  }, [open]);
+
+  const handleButtonClick = useCallback(() => {
+    setOpen(true);
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: I think this needs to be something like `togglePopoverVisibility` (or `setOpen(!open)`). As is, the dropdown doesn't close when the button is clicked if open, but it seems like it should?

Also, can you add whichever you use in here to the dependency array?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 5]
```diff
@@ -41,13 +51,31 @@ const PaginatedSelect: FunctionComponent<Props> = ({
...
+  }, []);
+
+  const handleOutsideClick = useCallback((event?: MouseEvent | undefined) => {
+    setOpen(false);
+  }, []);
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: Can you add `setOpen` to the dependency array? (or `setIsPopoverVisible` if you end up using the toggle state hook)

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 6]
```diff
@@ -41,13 +51,31 @@ const PaginatedSelect: FunctionComponent<Props> = ({
...
-      modifiers={{ arrow: { enabled: false }, offset: { offset: "0, 4" } }}
-      body={({ toggleVisibility }) => (
-        <ClickOutside onClickOutside={toggleVisibility}>
+    <ClickOutside onClickOutside={handleOutsideClick}>
+      <Popover
+        id=""
+        placement="bottom-end"
+        modifiers={{ arrow: { enabled: false }, offset: { offset: "0, 4" } }}
+        visible={open}
+        body={({ toggleVisibility }) => (
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: Since `toggleVisibility` is unused in this case now, you should be able to remove this and only pass through `IntersectionProvider` and everything inside of it.

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 7]
```diff
@@ -66,42 +94,61 @@ const PaginatedSelect: FunctionComponent<Props> = ({
...
+                  ref={filterRef}
+                  aria-roledescription="Filter results"
+                />
+              </Localized>
+            ) : (
+              <Flex
+                alignItems="center"
+                className={styles.wrapper}
+                style={{
+                  // display: open && !!onFilter ? "none" : "inherit",
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: Can this whole `style` prop just be removed now?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 8]
```diff
@@ -66,42 +94,61 @@ const PaginatedSelect: FunctionComponent<Props> = ({
...
+                }}
+                role="button"
+                tabIndex={0}
+                onFocus={() => setOpen(true)}
+                onBlur={() => setOpen(false)}
+              >
+                {selected}
+              </Flex>
+            )}
+            {!open && (
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: I know this is how it was before, but we could refactor these buttons to simplify and just check for visibility within the `Icon` prop and set the icon that way. For example, like
```
Icon={open ? ArrowsUpIcon : ArrowsDownIcon}
```

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx [hunk 9]
```diff
@@ -1,16 +1,18 @@
...
   useLoadMore,
+  useRefetch,
   withPaginationContainer,
 } from "coral-framework/lib/relay";
 
 import { SiteSelectorContainer_query } from "coral-admin/__generated__/SiteSelectorContainer_query.graphql";
 import { SiteSelectorContainer_viewer } from "coral-admin/__generated__/SiteSelectorContainer_viewer.graphql";
 import { SiteSelectorContainerPaginationQueryVariables } from "coral-admin/__generated__/SiteSelectorContainerPaginationQuery.graphql";
 
+import { useCoralContext } from "coral-framework/lib/bootstrap";
```
Comment (kabeaty) [inline src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx]: Can you please move this import up to the correct grouping?

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx [hunk 10]
```diff
@@ -22,7 +24,37 @@ interface Props {
 }
 
 const SiteSelectorContainer: React.FunctionComponent<Props> = (props) => {
+  const context = useCoralContext();
   const [loadMore, isLoadingMore] = useLoadMore(props.relay, 10);
+  // Buffering in the sense that we gather changes and then "flush"
+  // them 1 second after the last change is recieved
```
Comment (kabeaty) [inline src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx]: typo, `reieved` --> `received`

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx [hunk 11]
```diff
@@ -22,7 +24,37 @@ interface Props {
...
+  useEffect(() => {
+    if (filterBuffer === searchFilter) {
+      return;
+    }
+
+    if (searchTimeout) {
+      context.window.clearTimeout(searchTimeout);
+    }
+
+    const newTimeout = context.window.setTimeout(
```
Comment (kabeaty) [inline src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx]: Nice! This buffer works well when searching.

File: src/core/client/oembed/commentEmbed.html [hunk 12]
```diff
@@ -45,6 +46,7 @@ <h1>Comment Embed</h1>
           ><div style="margin-bottom: 8px;">Username</div><div>I am a comment. I am a comment. I am a comment. I am a comment. I am
           a comment. I am a comment. I am a comment. I am a comment. I am a comment.</div></div>
         </p> -->
+        <div class="coral-comment-embed coral-comment-embed-simple" style="background-color: #f4f7f7; padding: 8px; position: relative;" data-commentID=302a8724-599b-4751-a2bb-7f8f1bc3c2bd data-reactionLabel="Respect"><a target="_blank" style="position: absolute; z-index: 0; top: 0; bottom: 0; left: 0; right: 0;" href="http://localhost:8001/story/4?commentID=302a8724-599b-4751-a2bb-7f8f1bc3c2bd"></a><div style="margin-bottom: 8px;">admin</div><div><div>ive written a comment</div></div></div>
```
Comment (kabeaty) [inline src/core/client/oembed/commentEmbed.html]: Can you update to not commit this embedded comment please?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 13]
```diff
@@ -66,42 +94,61 @@ const PaginatedSelect: FunctionComponent<Props> = ({
...
+            className={cn(styles.button, className)}
+            onClick={handleButtonClick}
+            ref={ref}
+            justifyContent="space-between"
+          >
+            {Icon && (
+              <ButtonSvgIcon className={styles.buttonIconLeft} Icon={Icon} />
+            )}
+            {open && !!onFilter ? (
+              <Localized id="admin-paginatedSelect-filter">
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: I think the `aria-roledescription` also needs to be passed through in `attrs` here.

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 14]
```diff
@@ -66,42 +94,61 @@ const PaginatedSelect: FunctionComponent<Props> = ({
...
+                <TextField
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: The styles for the text input seem a bit different from the design; for example, it has a border when focused and the text looks darker. Should these styles be updated?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 15]
```diff
@@ -66,42 +94,61 @@ const PaginatedSelect: FunctionComponent<Props> = ({
...
+                  // display: open && !!onFilter ? "none" : "inherit",
+                  textOverflow: "ellipsis",
+                  whiteSpace: "nowrap",
+                }}
+                role="button"
+                tabIndex={0}
+                onFocus={() => setOpen(true)}
+                onBlur={() => setOpen(false)}
+              >
+                {selected}
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: In the dropdown, in the design, `All sites` has a separator beneath it, but I'm not seeing that here. Should the styles be updated to include this separator?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 16]
```diff
@@ -66,42 +94,61 @@ const PaginatedSelect: FunctionComponent<Props> = ({
...
-          uppercase={false}
-        >
-          {Icon && (
-            <ButtonSvgIcon className={styles.buttonIconLeft} Icon={Icon} />
-          )}
-          <Flex alignItems="center" className={styles.wrapper}>
-            {selected}
+        )}
+      >
+        {({ toggleVisibility, ref }) => (
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: I think you can just remove unused `toggleVisibility` here.

**Review by kabeaty — CHANGES_REQUESTED** [review]
This is working well overall! I have a few suggestions and comments that I left throughout. Also, did you test that the `PaginatedSelect` is still working as expected for the section selector? 

Also, can you add a test for this?

File: src/core/client/admin/components/AutoLoadMore.tsx (hunk 1 above)
Comment (marcushaddon) [reply src/core/client/admin/components/AutoLoadMore.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 2 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 3 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅ I've implemented what I'm pretty sure you mean.

Do you mean that the popover should close when the user clicks on a text input that has focus (which is what the button becomes when the popover opens, given that the onFilter callback is provided)?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 4 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 5 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 6 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 7 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 8 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx (hunk 9 above)
Comment (marcushaddon) [reply src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx]: ✅

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx (hunk 10 above)
Comment (marcushaddon) [reply src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx]: ✅

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx (hunk 11 above)
Comment (marcushaddon) [reply src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx]: ✅ ty!

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 13 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 16 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/oembed/commentEmbed.html (hunk 12 above)
Comment (marcushaddon) [reply src/core/client/oembed/commentEmbed.html]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 14 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅  I've updated the color of the text, however I think removing the outline on focus is an accessibility violation: https://www.a11yproject.com/posts/never-remove-css-outlines/#:%7E:text=Using%20the%20CSS%20rule%20%3Afocus,with%20the%20link%20or%20control.

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 15 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: (this file won't say "outdated" because the change was in SiteSelector, not PaginatedSelect)

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx [hunk 17]
```diff
@@ -22,7 +24,37 @@ interface Props {
...
+      () => setSearchFilter(filterBuffer),
+      1000
+    );
+
+    setSearchTimeout(newTimeout);
+
+    return () => context.window.clearTimeout(searchTimeout);
+  }, [filterBuffer, context.window]);
+  // marcushaddon: omitting searchFilter from dependency array to avoid an infinite update loop
```
Comment (nick-funk) [inline src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx]: Thanks for handling this edge case!

I noticed something else going on if the query to filter/paginate errors. When there is an error, the page spams out thousands of requests as fast as possible, hammering the API.

A good way to test this is without an index for Site (maybe missing in PR description notes, maybe an optional from `INDEXES.md`?) try and filter. Mongo will error cause it can't do text searches without a text index. Open the dev tools console and you'll see the page is hammering the API with retries to filter the site list.

If possible, can we throttle retries or limit retries on these filter requests?

**Review by nick-funk — CHANGES_REQUESTED** [review]

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 15 above)
Comment (kabeaty) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: thanks for adding! It looks like it has more spacing than the design, and maybe the spacing should be more like 2? Can you take a look?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 6 above)
Comment (kabeaty) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: thanks for updating! I think you can go one step further and just make the body the `IntersectionProvider` component and everything inside of it now.

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 7 above)
Comment (kabeaty) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: thanks for updating! Can the `textOverflow` and `whiteSpace` styles also be removed? Looks like they're both here and in the css?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 18]
```diff
@@ -10,18 +16,21 @@ import {
   ButtonSvgIcon,
 } from "coral-ui/components/icons";
 import {
-  Button,
   ClickOutside,
   Dropdown,
   Flex,
   Popover,
   Spinner,
+  // TextField,
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: Can you please remove this?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 19]
```diff
@@ -10,18 +16,21 @@ import {
...
 } from "coral-ui/components/v2";
 
+import { useToggleState } from "coral-framework/hooks";
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: Can you please move these imports up to the correct grouping?

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelector.tsx [hunk 20]
```diff
@@ -9,6 +9,7 @@ import { AppWindowIcon } from "coral-ui/components/icons";
 import SiteSelectorCurrentSiteQuery from "./SiteSelectorCurrentSiteQuery";
 import SiteSelectorSite from "./SiteSelectorSite";
 
+import { Divider } from "coral-ui/components/v2";
```
Comment (kabeaty) [inline src/core/client/admin/routes/Moderate/SiteSelector/SiteSelector.tsx]: Can you please move this to the correct import grouping?

File: src/core/client/ui/components/v2/Divider/Divider.css [hunk 21]
```diff
@@ -7,17 +7,26 @@ $divider-border-color: var(--palette-grey-300);
...
+  margin-bottom: var(--spacing-3);
 }
 
 .spacing4 {
-  margin: var(--spacing-4) 0;
+  margin-top: var(--spacing-4);
+  margin-bottom: var(--spacing-4);
+}
+
+.horizontallySpaced {
```
Comment (kabeaty) [inline src/core/client/ui/components/v2/Divider/Divider.css]: Would it be more flexible for future usage to make this a spacing that can be passed in that defaults to no spacing if nothing is passed in? So, like, `horizontalSpacing={2}` and then that providers the left and right spacing of 2, but then if nothing gets passed in for it, it defaults to 0?

**Review by kabeaty — CHANGES_REQUESTED** [review]
thanks for making updates! I left a few more small comments/questions. Also, wanted to follow up on my previous review comment and check if you tested that the `PaginatedSelect` is still working as expected for the section selector with these updates? Also, did you look into adding a test?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 18 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 19 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelector.tsx (hunk 20 above)
Comment (marcushaddon) [reply src/core/client/admin/routes/Moderate/SiteSelector/SiteSelector.tsx]: ✅

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx (hunk 17 above)
Comment (marcushaddon) [reply src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx]: Do we have the ability to specify a rate limit on the front end using PaginationContainers? I'm not seeing any examples of doing this on a per-request basis.

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 15 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: Agreed, 1 actually looked closer to me. Lemme know if you disagree!

File: src/core/client/ui/components/v2/Divider/Divider.css (hunk 21 above)
Comment (marcushaddon) [reply src/core/client/ui/components/v2/Divider/Divider.css]: ✅

**Comment by marcushaddon** [comment]
> This is working well overall! I have a few suggestions and comments that I left throughout. Also, did you test that the `PaginatedSelect` is still working as expected for the section selector?
> 
> Also, can you add a test for this?

@kabeaty The SectionSelector is working, and I've added a test for the SiteSelector

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 15 above)
Comment (kabeaty) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: looks lovely! thanks for updating

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 22]
```diff
@@ -1,27 +1,37 @@
...
 import {
-  Button,
   ClickOutside,
   Dropdown,
   Flex,
   Popover,
   Spinner,
 } from "coral-ui/components/v2";
 
+import { TextArea } from "coral-ui/components/v3";
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: Can you move this import up to the correct section?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 23]
```diff
@@ -33,21 +43,41 @@ interface Props {
...
+      filterRef.current.focus();
+    }
+  }, [isPopoverVisible]);
+
+  const handleButtonBlur = useCallback(() => {
+    if (!onFilter) {
+      return;
+    }
+    setIsPopoverVisible(false);
+  }, [onFilter]);
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: Can you include `setIsPopoverVisible` in the dep array here please?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 24]
```diff
@@ -66,42 +96,58 @@ const PaginatedSelect: FunctionComponent<Props> = ({
...
-            <ButtonSvgIcon className={styles.buttonIconLeft} Icon={Icon} />
-          )}
-          <Flex alignItems="center" className={styles.wrapper}>
-            {selected}
+        )}
+      >
+        {({ ref }) => (
+          <Flex
+            className={cn(styles.button, className)}
+            onClick={() => setIsPopoverVisible(true)}
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: It looks like now if you click this arrow icon button on the site selector, it doesn't close the popover as I would expect it to (and as it was before this latest update; I'm guessing this was part of the updates to address the section selector functionality). Can you take a look?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 25]
```diff
@@ -66,42 +96,58 @@ const PaginatedSelect: FunctionComponent<Props> = ({
...
+            justifyContent="space-between"
+            aria-label={label}
+          >
+            {Icon && (
+              <ButtonSvgIcon className={styles.buttonIconLeft} Icon={Icon} />
+            )}
+            {isPopoverVisible && !!onFilter ? (
+              <Localized
+                id="admin-paginatedSelect-filter"
+                attrs={{ "aria-roledescription": true, "aria-label": true }}
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: Looks like this `Localized` is only for the `TextArea` which only has an `aria-label` attribute and not an `aria-roledescription` (that's the button below)? Can you take a look?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 26]
```diff
@@ -66,42 +96,58 @@ const PaginatedSelect: FunctionComponent<Props> = ({
...
+                  onChange={(e) => onFilter(e.target.value)}
+                  ref={filterRef}
+                  aria-label="Filter results"
+                />
+              </Localized>
+            ) : (
+              <Flex
+                alignItems="center"
+                className={styles.wrapper}
+                style={{
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: Is this `style` still needed here since it's also in the css?

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx [hunk 27]
```diff
@@ -33,21 +43,41 @@ interface Props {
...
-      modifiers={{ arrow: { enabled: false }, offset: { offset: "0, 4" } }}
-      body={({ toggleVisibility }) => (
-        <ClickOutside onClickOutside={toggleVisibility}>
+    <ClickOutside onClickOutside={() => setIsPopoverVisible(false)}>
+      <Popover
+        id=""
+        placement="bottom-end"
+        modifiers={{ arrow: { enabled: false }, offset: { offset: "0, 4" } }}
+        visible={isPopoverVisible}
+        body={() => (
```
Comment (kabeaty) [inline src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: I think you can just make the body the `IntersectionProvider` component and everything inside of it now.

**Review by kabeaty — CHANGES_REQUESTED** [review]
Looking good! thanks for adding the tests and checking on the section selector functionality! Just left a few more comments/questions related to the updates.

File: src/core/client/admin/test/moderate/regularQueue.spec.tsx [hunk 28]
```diff
@@ -813,6 +816,133 @@ it("rejects comment in reported queue", async () => {
...
+  const siteSelector = await within(searchBarContainer).findByLabelText(
+    "Select site",
+    { exact: false }
+  );
+  expect(siteSelector).toBeInTheDocument();
+
+  userEvent.click(siteSelector);
+
+  await waitFor(() =>
+    expect(within(siteSelector).getByLabelText("Filter results")).toBeVisible()
```
Comment (kabeaty) [inline src/core/client/admin/test/moderate/regularQueue.spec.tsx]: It looks like the test is currently failing because you're looking for `Filter results` here but in localization `results` is capitalized (`Filter Results`). Also, you should be able to simplify here to just `expect(await within(siteSelector).findByLabelText("Filter Results")).toBeVisible();` if you like.

File: src/core/client/admin/test/moderate/regularQueue.spec.tsx [hunk 29]
```diff
@@ -813,6 +816,133 @@ it("rejects comment in reported queue", async () => {
...
+  );
+
+  const dropDown = screen.getByRole("dialog");
+  expect(dropDown).toBeVisible();
+
+  const textInput = await within(siteSelector).findByLabelText(
+    "Filter results",
```
Comment (kabeaty) [inline src/core/client/admin/test/moderate/regularQueue.spec.tsx]: I think you can just search for `Filter Results` here without the `exact: false` too

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 22 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 23 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 25 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 27 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 26 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/test/moderate/regularQueue.spec.tsx (hunk 28 above)
Comment (marcushaddon) [reply src/core/client/admin/test/moderate/regularQueue.spec.tsx]: ✅

File: src/core/client/admin/test/moderate/regularQueue.spec.tsx (hunk 29 above)
Comment (marcushaddon) [reply src/core/client/admin/test/moderate/regularQueue.spec.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 24 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 6 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 7 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: ✅

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 26 above)
Comment (kabeaty) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: I think the `className={styles.wrapper}` needs to be added back in here now

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx [hunk 30]
```diff
@@ -1,16 +1,17 @@
...
   useLoadMore,
+  useRefetch,
   withPaginationContainer,
 } from "coral-framework/lib/relay";
 
 import { SiteSelectorContainer_query } from "coral-admin/__generated__/SiteSelectorContainer_query.graphql";
 import { SiteSelectorContainer_viewer } from "coral-admin/__generated__/SiteSelectorContainer_viewer.graphql";
 import { SiteSelectorContainerPaginationQueryVariables } from "coral-admin/__generated__/SiteSelectorContainerPaginationQuery.graphql";
-
 import SiteSelector from "./SiteSelector";
```
Comment (kabeaty) [inline src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx]: Can you add the space separating these import groups back in please?

File: src/core/client/ui/components/v2/Flex/Flex.tsx [hunk 31]
```diff
@@ -105,7 +105,7 @@ const Flex: FunctionComponent<Props> = (props) => {
   return (
     <Box
       ref={forwardRef}
-      className={cn(rootClassNames, flexClassNames)}
+      className={cn(rootClassNames, flexClassNames, "hahahaha")}
```
Comment (kabeaty) [inline src/core/client/ui/components/v2/Flex/Flex.tsx]: Looks like this snuck in...can you remove :)?

**Review by kabeaty — CHANGES_REQUESTED** [review]
thanks for the updates! looking great. Just two small comments.

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx (hunk 30 above)
Comment (marcushaddon) [reply src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx]: ✅

File: src/core/client/ui/components/v2/Flex/Flex.tsx (hunk 31 above)
Comment (marcushaddon) [reply src/core/client/ui/components/v2/Flex/Flex.tsx]: ✅

File: src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx (hunk 17 above)
Comment (nick-funk) [reply src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx]: Yeah, if you look into the `AutoLoadMore` there is a `useEffect` that is performing the `onLoadMore`. Could stick a retry counter or use `setTimeOut`'s to back off the reloads I think? If that doesn't work, then something higher up is likely re-calling the load events and you can back it off there instead?

**Review by kabeaty — APPROVED** [review]
looks great! thanks for the updates!

File: src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx (hunk 26 above)
Comment (marcushaddon) [reply src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx]: I've replaced it, but just a heads up, the ellipses overflow doesn't work, and wasn't working before this PR because of of the way that the `<Flex>` component wraps its contents in an extra div. It's worth a fix but since it is a problem that has existed for along time I think it is outside of the scope of this ticket.

**Review by nick-funk — APPROVED** [review]
Looks good!
